pytest-selenium/tests/test_search.py

```diff
@@ -0,0 +1,92 @@
+from random import choice
+from string import digits, ascii_letters
+
+from pages.content import Content
+from tests import markers
+
+
+@markers.test_case("C543235")
+@markers.parametrize("page_slug", ["preface"])
+@markers.nondestructive
+def test_message_when_search_yields_no_results(selenium, base_url, book_slug, page_slug):
+    # GIVEN: Book page is loaded
+    content = Content(selenium, base_url, book_slug=book_slug, page_slug=page_slug).open()
+    toolbar = content.toolbar
+    mobile = content.mobile_search_toolbar
+    search_sidebar = content.search_sidebar
+    page_url_before_search = content.current_url
+
+    # Using regex to create a random search term
+    search_term = "".join(choice(digits + ascii_letters) for i in range(25))
+
+    # WHEN: Search is triggered for a term which yields no results
+    if content.is_desktop:
+        toolbar.search_for(search_term)
+
+    if content.is_mobile:
+        mobile.search_for(search_term)
+
+    # THEN: Search sidebar displays no results message
+    assert (
+        search_sidebar.no_results_message == "Sorry, no results found for   ‘"
+        "{}"
+        "’".format(search_term)
```

Comment:
Maybe replace the string with a formatted string to condense the statement:
`f"Sorry, no results found for   ‘{search_term}’"`

Reply:
Done